pkg/provider/akeyless/akeyless.go

```diff
@@ -71,6 +71,33 @@ func (p *Provider) NewClient(ctx context.Context, store esv1beta1.GenericStore,
 }
 
 func (p *Provider) ValidateStore(store esv1beta1.GenericStore) error {
+	storeSpec := store.GetSpec()
+	akeylessSpec := storeSpec.Provider.Akeyless
+
+	akeylessGWApiURL := akeylessSpec.AkeylessGWApiURL
+
+	if akeylessGWApiURL == nil {
```

Comment:
We can also check if this is a valid URL with url.Parse. there are some other providers doing that. Senhasegura and the 1pass PR for instance :)

Reply:
Also, we only check if the field is nill in a very specific case, afaik. Only when we have two fields that are optional but we need at least one of them to be passed.

Isn't that correct, @gusfcarvalho @moolen ? Because kubebuilder already tags these fields as required for us.

Reply:
Actually, we cannot have this check here, see https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/akeyless/akeyless.go#L88-L91

Reply:
@gusfcarvalho Are you saying the check it not needed in `ValidateStore` since we already have this check [here](https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/akeyless/akeyless.go#L88-L91)?

Reply:
This URL has a default value. So it can, and most likely will be empty by the user.

But we can check (if it is not empty) if it is a valid URL with url.Parse